lms/djangoapps/mobile_api/mobile_platform.py

```diff
@@ -63,7 +63,7 @@ class IOS(MobilePlatform):
 
 class Android(MobilePlatform):
     """ Android platform """
-    USER_AGENT_REGEX = (r'Dalvik/[.0-9]+ \(Linux; U; Android [.0-9]+; (.*) Build/[0-9a-zA-Z]*\) '
+    USER_AGENT_REGEX = (r'Dalvik/[.0-9]+ \(Linux; U; Android [.0-9]+; (.*) (Build|MIUI)/[0-9a-zA-Z-.]*\) '
```

Comment:
regex is smart enough to know that the dash after the end of a range sequence is just a dash and not a new range? Clever things.

Reply:
Yeah, it's magical.

Reply:
Btw is this part `(.*) (Build|MIUI)/[0-9a-zA-Z-.]*\)` of the regex really necessary? How about just checking if one-to-many characters of any form are available in-between `;` and `)`?

**_Its highly unlikely_**, but there might come a use cases where a vendor decides to mess up this part of their User-Agent like:
```
Dalvik/2.1.0 (Linux; U; Android 5.1; Phatom Z Whatever/Anything) edX/org.edx.mobile/2.2.2
```

Reply:
So honestly, I'm not entirely sure. In this work, I tried to continue to be as restrictive as possible and if a new bug ticket is filed to be more permissive, approaching that problem then.
In all of this work, I am happy to defer to the mobile team on the desired functionality. As part of the [ticket](https://openedx.atlassian.net/browse/AA-324), I was given a list of failing examples and updated the regex to handle those situations which is how I came to the regex in this PR.

Reply:
Makes sense, thanks! 👍